src/bootstrap/dist.rs

```diff
@@ -362,6 +362,11 @@ impl Step for Rustc {
 
             builder.install(&builder.rustdoc(compiler), &image.join("bin"), 0o755);
 
+            let ra_proc_macro_srv = builder
+                .ensure(tool::RustAnalyzerProcMacroSrv { compiler, target: compiler.host })
+                .expect("rust-analyzer-proc-macro-server always builds");
+            builder.install(&ra_proc_macro_srv, &image.join("libexec"), 0o755);
```

Comment:
So I think the problem here is that this is using the `compiler` that this step uses/produces (`run.builder.compiler(run.builder.top_stage, run.target)`), which isn't quite right for compiling other things. (Since it means that we need to build a std using that compiler, which isn't otherwise typically necessary).

I think the fix here is going to be to change compiler in the tool invocation to be closer to what is used for e.g. Miri compilation elsewhere -- with slight modifications: `builder.compiler_for(compiler.stage, builder.config.build, compiler.host)` looks right to me. My hope is that with that change the tests will no longer need modification.

Reply:
That seems to have done it! https://github.com/rust-lang/rust/pull/99756/commits/4c4ea87729cbd964e4263499c41149d24c8cbb00 applies your suggested changes, and https://github.com/rust-lang/rust/pull/99756/commits/55b8ef4d9a4024e638aa71ffa1dfedcff41081fb reverses the test changes.